sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitions.scala

```diff
@@ -39,31 +72,35 @@ private[sql] object PruneFileSourcePartitions extends Rule[LogicalPlan] {
             _,
             _))
         if filters.nonEmpty && fsRelation.partitionSchemaOption.isDefined =>
-      val normalizedFilters = DataSourceStrategy.normalizeExprs(
-        filters.filterNot(SubqueryExpression.hasSubquery), logicalRelation.output)
-
-      val sparkSession = fsRelation.sparkSession
-      val partitionColumns =
-        logicalRelation.resolve(
-          partitionSchema, sparkSession.sessionState.analyzer.resolver)
-      val partitionSet = AttributeSet(partitionColumns)
-      val partitionKeyFilters = ExpressionSet(normalizedFilters.filter { f =>
-        f.references.subsetOf(partitionSet)
-      })
-
+      val partitionKeyFilters = getPartitionKeyFilters(
+        fsRelation.sparkSession, logicalRelation, partitionSchema, filters, logicalRelation.output)
       if (partitionKeyFilters.nonEmpty) {
         val prunedFileIndex = catalogFileIndex.filterPartitions(partitionKeyFilters.toSeq)
         val prunedFsRelation =
-          fsRelation.copy(location = prunedFileIndex)(sparkSession)
+          fsRelation.copy(location = prunedFileIndex)(fsRelation.sparkSession)
```

Comment:
I suggest to pass also the `dataFilters`.
This is useful for FileIndex implementations that use the `dataFilters` to do the file listing.
For example, we at IBM Research use this to provide data skipping for all file based datasources.
I suggest something like this guykhazma@ de3415b80b92e635dbc0b331515a8cec4c5098d6

Reply:
@guykhazma Thanks for the suggestion.
However, the `PartitioningAwareFileIndex` doesn't use the data filters for listing files. Could you provide an example that the data filters will be useful here?
Also, the data filters are supposed to be pushed down in `FileScanBuiler` (e.g ORC/Parquet)

Reply:
@gengliangwang this is useful for enabling data skipping on all file formats includes formats which doesn't support pushdown (e.g CSV, JSON) by replacing the FileIndex implementation with a FileIndex which use also the `dataFilters` to filter the file listing.

Reply:
This is the old v1 code path, let's not touch it in this PR.

Reply:
and for v2 code path, the data filters are already pushed in the rule `V2ScanRelationPushDown`